### PR TITLE
Invade static properties and methods

### PR DESCRIPTION
**Pull request by otsch**
Add `invadeStatic()` allowing to also invade private static properties and methods.
I suggested this a while ago [on twitter](https://twitter.com/chrolear/status/1780891662232789306)

----------

__From the added README:__

Further, there is also `invadeStatic` to get and set private static class properties and call private static methods. Imagine having this class:

```php
class MyClass
{
    private static string $privateProperty = 'privateValue';

    private static function privateMethod(string $string, int $int): string
    {
        return 'private return value ' . $string . ' ' . $int;
    }
}
```

Here is how you get and set private class properties:

```php
invadeStatic(MyClass::class)->get('privateProperty'); // returns 'private value'

invadeStatic(MyClass::class)->set('privateProperty', 'changedValue');

invadeStatic(MyClass::class)->get('privateProperty'); // returns 'changedValue'
```

And this is how you call private static methods:

```php
invadeStatic(MyClass::class)
    ->method('privateMethod')
    ->call('foo', 123);

// returns 'private return value foo 123'
```

### CODE REVIEW

**Comment by freekmurze** [comment]
Seems like the tests are failing. Could you have a look?

**Comment by freekmurze** [comment]
Could we somehow let `invade` detect whether a property function is static or not and do the right thing? That way we would have the user care about choosing between `invade` and `invadeStatic`

File: README.md [hunk 1]
```diff
@@ -78,17 +78,17 @@ class MyClass
...
-invadeStatic(MyClass::class)->get('privateProperty'); // returns 'changedValue'
+invade(MyClass::class)->get('privateProperty'); // returns 'changedValue'
 ```
 
 And this is how you call private static methods:
 
 ```php
-invadeStatic(MyClass::class)
+invade(MyClass::class)
     ->method('privateMethod')
```
Comment (freekmurze) [inline README.md:92]: for clarity, use the string `'privateStaticMethod'`

File: src/functions.php [hunk 2]
```diff
@@ -7,22 +7,15 @@
...
+     * @param T|class-string $object
+     * @return Invader<T>|StaticInvader
      */
-    function invade(object $object): Invader
+    function invade(object|string $object): Invader|StaticInvader
     {
-        return new Invader($object);
-    }
-}
+        if (is_object($object)) {
```
Comment (freekmurze) [inline src/functions.php:15]: Nice!

**Comment by otsch** [comment]
> Could we somehow let `invade` detect whether a property function is static or not and do the right thing? That way we would have the user care about choosing between `invade` and `invadeStatic`

I tried that at first yesterday, but had some trouble getting the phpdocs right. But I got it now: https://github.com/spatie/invade/pull/32/commits/c86f3807d6ff9f06ada5ddac0c04eea721d28cc0 
It decides wheter to use the `Invader` or `StaticInvader` based on the argument being a class name string or an object.

----------

Tests should also be fixed now: https://github.com/spatie/invade/pull/32/commits/cc439e457b22d2906082190ff17f229ca5dfc377 

with `--prefer-lowest` and PHP 8.3 locally, I also get: 
```
Creation of dynamic property P\Tests\InvaderTest::$class is deprecated
```
errors in `tests/InvaderTest.php`. But I think that relies on a fix in pest or somewhere (that's probably only missing in versions installed with `--prefer-lowest`).

File: README.md (hunk 1 above)
Comment (otsch) [reply README.md]: 👍🏻 changed it: https://github.com/spatie/invade/pull/32/commits/60173da4baf4ad91e0a13f1ac7c5d1c507883885

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**